components/admin/lessons/AdminLessonInfo.tsx

```diff
@@ -32,7 +32,7 @@ type NewLessonProps = {
 
 // Creates card for a lessons's information to update
 const EditLesson: React.FC<EditLessonProps> = ({ setLessons, lesson }) => {
-  const [alterLesson, { loading, data }] = useMutation(updateLesson)
+  const [alterLesson, { loading, data, error }] = useMutation(updateLesson)
```

Comment:
Could take this opportunity to also refactor and use the auto-generated typed mutation.

```jsx
import { useUpdateLessonMutation } from '../../../graphql/index'

const [alterLesson, { loading, data, error }] = useUpdateLessonMutation()
```

Reply:
I'm running into a type issue when implimenting this refactor.
```
Argument of type 'Maybe<Maybe<{ __typename?: "Lesson" | undefined; id: number; docUrl?: Maybe<string> | undefined; githubUrl?: Maybe<string> | undefined; videoUrl?: Maybe<string> | undefined; ... 5 more ...; challenges: { ...; }[]; }>[]> | undefined' is not assignable to parameter of type 'SetStateAction<Lesson[] | null>'.
  Type 'undefined' is not assignable to type 'SetStateAction<Lesson[] | null>'.ts(2345)
```

I can solve it with casting the value but that seems like the wrong thing to do here 
```
    !loading && data && setLessons(data.updateLesson as Lesson[])
```

Reply:
Just another victim of mistaken type on the graphql schema. Change the response type for this mutation to return `[Lesson!]!` on `graphql/typeDefs.ts`

```graphql
updateLesson(
 # Mutation args...
): [Lesson!]!
```

Reply:
Maybe have a backlog issue where someone reviews all the types and make sure all the non-nullable values are set so it doesn't clash with typescript.

Reply:
It also seems like we have the graphql codegen generating `typescript-resolvers` types but then we are not actually using them. Maybe if we wire those to the resolvers we will discover all the type mismatches we have.

Reply:
Updated the typedefs and swapped in the new codegen mutations d4a97de

Reply:
> It also seems like we have the graphql codegen generating typescript-resolvers types but then we are not actually using them. Maybe if we wire those to the resolvers we will discover all the type mismatches we have.

That's true, I've added some of those return types in the resolvers I worked with in the past. But there's some issues that may happen and it's not about the code being wrong, for example, TS complains about a field with a `Date` type not being a `string` which is expected on the return type, but that conversion is handled automatically by the GraphQL server.